### ISSUE TITLE
ProtectKeysWithCertificate Key {elided} is ineligible to be the default key because its CreateEncryptor method failed.

### ISSUE THREAD

**Issue by jons-aura** [aspnet/AspNetCore#12523]
### Describe the bug
I have a .Net Core 2.2 application that I am trying to have persist the session keys to disk so that future apps and future updates to existing apps for a particular client can use them for single sign on.

The application works fine on my development machine. When deployed to our test environment I get the error below. As far as I can tell the cert is found, and the key is written successfully as XML appears on disk with data marked as encrypted in it. If I remove the "ProtectKeysWithCertificate" call everything works but the XML key data file on disk is as expected in that case not encrypted.

### To Reproduce
1. Using ASP.NET Core 2.2
2. Run this code
````

X509Store store = new X509Store(StoreLocation.LocalMachine);
store.Open(OpenFlags.ReadOnly);
X509Certificate2Collection x509Certificate2Collection = store.Certificates.Find(X509FindType.FindByThumbprint, GetConfigurationValue<string>(ConfigurationKeys.SessionKeysEncryptionCertThumbprint), false);
X509Certificate2 x509Cert = x509Certificate2Collection.Count > 0 ? x509Certificate2Collection[0] : null;
store.Close();

services.AddDataProtection()
    .SetApplicationName("APPNAME")
    .ProtectKeysWithCertificate(x509Cert)
    .PersistKeysToFileSystem(new DirectoryInfo(GetConfigurationValue<string>(ConfigurationKeys.SessionKeysPath)));
````
3. The certificate is in the local machine store.
````
fail: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[24]
      An exception occurred while processing the key element '<key id="2d0e4543-857d-4e47-a3c1-fe82a552a75f" version="1" />'.
Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException: The system cannot find the file specified
   at System.Security.Cryptography.CngKey.Open(String keyName, CngProvider provider, CngKeyOpenOptions openOptions)
   at System.Security.Cryptography.CngKey.Open(String keyName, CngProvider provider)
   at Internal.Cryptography.Pal.CertificatePal.GetPrivateKey[T](Func`2 createCsp, Func`2 createCng)
   at Internal.Cryptography.Pal.CertificatePal.GetRSAPrivateKey()
   at Internal.Cryptography.Pal.CertificateExtensionsCommon.GetPrivateKey[T](X509Certificate2 certificate, Predicate`1 matchesConstraints)
   at System.Security.Cryptography.X509Certificates.RSACertificateExtensions.GetRSAPrivateKey(X509Certificate2 certificate)
   at Microsoft.AspNetCore.DataProtection.XmlEncryption.EncryptedXmlDecryptor.EncryptedXmlWithCertificateKeys.GetKeyFromCert(EncryptedKey encryptedKey, KeyInfoX509Data keyInfo)
   at Microsoft.AspNetCore.DataProtection.XmlEncryption.EncryptedXmlDecryptor.EncryptedXmlWithCertificateKeys.DecryptEncryptedKey(EncryptedKey encryptedKey)
   at System.Security.Cryptography.Xml.EncryptedXml.GetDecryptionKey(EncryptedData encryptedData, String symmetricAlgorithmUri)
   at System.Security.Cryptography.Xml.EncryptedXml.DecryptDocument()
   at Microsoft.AspNetCore.DataProtection.XmlEncryption.EncryptedXmlDecryptor.Decrypt(XElement encryptedElement)
   at Microsoft.AspNetCore.DataProtection.XmlEncryption.XmlEncryptionExtensions.DecryptElement(XElement element, IActivator activator)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager.Microsoft.AspNetCore.DataProtection.KeyManagement.Internal.IInternalXmlKeyManager.DeserializeDescriptorFromKeyElement(XElement keyElement)
warn: Microsoft.AspNetCore.DataProtection.KeyManagement.DefaultKeyResolver[12]
      Key {2d0e4543-857d-4e47-a3c1-fe82a552a75f} is ineligible to be the default key because its CreateEncryptor method failed.
````

### Expected behavior
The application should be able to use the key it created.

**Comment by blowdart** [comment]
It appears GetPrivateKey() is failing.

On the machine where it fails on is the private key for the certificate present? 
Does the user account the app runs under have access to the private key in the cert store?

**Comment by jons-aura** [comment]
Works now. It appears that the app pool for this app did not have access to the private key. I granted it access and it appears to work. The error message and timing of the error seems rather misleading to me as I saw the data on disk so I didn't think it was the file not found issue and the CreateEncryptor error message didn't have any search results that I found that mentioned permissions.

It probably wouldn't hurt to add a note about permissions or a link to other documentation that notes permissions will need to be set to this page https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/implementation/key-encryption-at-rest?view=aspnetcore-3.0

Thanks for the quick response!
